src/model/include/iDynTree/ModelTestUtils.h

```diff
@@ -211,7 +239,10 @@ inline bool getRandomInverseDynamicsInputs(FreeFloatingPos& pos,
     vel.baseVel() =  getRandomTwist();
     acc.baseAcc() =  getRandomTwist();
 
-
+    // Use model-aware joint position generation instead of naive random vector
+    // This is needed for proper handling of RevoluteSO2Joint and other special joints
+    // However, we don't have access to the model here, so we can't use getRandomJointPositions
+    // This function is not used in the failing test anyway, but should be addressed separately
```

Comment:
Probably we should modify `getRandomInverseDynamicsInputs` to pass also the `iDynTree::Model` ? If we do not want to solve this here, we should open an issue, as it is really diffiult to avoid loosing track of a problem reported just with just a comment in the code.